### PR TITLE
Algoo ooxml

### PR DESCRIPTION
**Pull request by K-Kumar-01**
### Changes

- <ONE> Adds the transformers for converting text and emphasize text (CiceroMarkJSON -> OOXML)
- <TWO> Adds the transformer for converting emphasize text(OOXML -> CiceroMarkJSON)
- <THree> Test created for the above conversions.

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [x] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Extend the documentation, if necessary
- [ ] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Comment by K-Kumar-01** [comment]
@algomaster99 
I have made a PR. Let me know if there are any changes required.

Also a thing I want to ask. After the previous commit(June-5) was merged, I did the following:
```
  git checkout master
  git fetch --all --prune
  git rebase upstream/master
  git push origin master
  ```
  
  And also `git checkout algoo-ooxml and git rebase master`. However, the old commits are still visible. 
  Is there something wrong which I followed. 
  Thanks in advance :)

**Comment by algomaster99** [comment]
> Is there something wrong which I followed.

Yes. `git fetch` only gets the new references but **does not update** your local repository. You need to merge it also. I advise you to instead use `git pull` which will `merge` your changes too.

I also want to bring a few more things related to your attention.

- Is there a reason why you have named your branch `algo-ooxml`? AccordProject has [laid out guidelines](https://github.com/accordproject/techdocs/blob/master/CONTRIBUTING.md#-pull-request-submission-guidelines) so you _may_ follow them. However, the branch is anyway going to be deleted (no point to keep them after they are merged), so you may not _strictly_ follow them but I would at least expect a short description there.
- PR title is also very random. Your title should be very clear about what it does.
- Link issues in your PR. It makes it easier to navigate across issues/pull requests. For example, #397 should be linked in the PR description.

I get that all of this may sound time-consuming and you already have a lot to code but you need to understand that reviewing is equally tough if not more. Better PR title and better description tell the reviewer about what they can expect from the code _beforehand_. Thus, making it easier for one to review and perhaps, add more suggestions which would be a good thing for you.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+const expect = chai.expect;
+
+const OoxmlTransformer = require('./OoxmlTransformer');
+const CiceroMarkToOOXMLTransfomer = require('./CiceroMarkToOOXMLTransformer');
+
+describe('Round Tripping', () => {
+
+    it('Pargaphs and emphasis', async () => {
+        const ciceroMarkJSON = {
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js:29]: You can put this entire JSON object in the test resources directory.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+'use strict';
+
+const chai = require('chai');
+
+const expect = chai.expect;
+
+const OoxmlTransformer = require('./OoxmlTransformer');
+const CiceroMarkToOOXMLTransfomer = require('./CiceroMarkToOOXMLTransformer');
+
+describe('Round Tripping', () => {
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js:26]: You should be descriptive here. Maybe something like "perform roundtripping between CiceroMark and OOXML ". I can give you a reason for this as well. When you run the entire test suite, and suppose this fails, you will have this name handy to know where it failed.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    it('Pargaphs and emphasis', async () => {
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.test.js:28]: Reword this to `should parse paragraphs and emphasis nodes`.

Also, there is a typographical error in "paragraphs". If you are prone to them, I suggest you to install a VSCode extension, or if your computer specification allows, switch to an IDE like WebStorm.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 4]
```diff
@@ -14,23 +14,134 @@
...
+            heading: 'org.accordproject.commonmark.Heading',
+            item: 'org.accordproject.commonmark.Item',
+            list: 'org.accordproject.commonmark.List',
+            listBlock: 'org.accordproject.ciceromark.ListBlock',
+            paragraph: 'org.accordproject.commonmark.Paragraph',
+            softbreak: 'org.accordproject.commonmark.Softbreak',
+            text: 'org.accordproject.commonmark.Text',
+            variable: 'org.accordproject.ciceromark.Variable',
+            emphasize: 'org.accordproject.commonmark.Emph',
+        };
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:39]: Move this out of the class if eslint cannot parse this as a class variable.

File: packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js [hunk 5]
```diff
@@ -14,23 +14,134 @@
...
+        }
+        return '';
+    }
+
+    /**
+     * Wraps OOXML in docx headers.
+     *
+     * @param {string} ooxml OOXML to be wrapped
+     */
+    wrapOOXML(ooxml){
```
Comment (algomaster99) [inline packages/markdown-docx/src/CiceroMarkToOOXMLTransformer.js:93]: These are exactly the kind of methods that should be in `helpers.`/`utils`.

**Comment by K-Kumar-01** [comment]
> > Is there something wrong which I followed.
> 
> Yes. `git fetch` only gets the new references but **does not update** your local repository. You need to merge it also. I advise you to instead use `git pull` which will `merge` your changes too.
> 
> I also want to bring a few more things related to your attention.
> 
> * Is there a reason why you have named your branch `algo-ooxml`? AccordProject has [laid out guidelines](https://github.com/accordproject/techdocs/blob/master/CONTRIBUTING.md#-pull-request-submission-guidelines) so you _may_ follow them. However, the branch is anyway going to be deleted (no point to keep them after they are merged), so you may not _strictly_ follow them but I would at least expect a short description there.
> * PR title is also very random. Your title should be very clear about what it does.
> * Link issues in your PR. It makes it easier to navigate across issues/pull requests. For example, #397 should be linked in the PR description.
> 
> I get that all of this may sound time-consuming and you already have a lot to code but you need to understand that reviewing is equally tough if not more. Better PR title and better description tell the reviewer about what they can expect from the code _beforehand_. Thus, making it easier for one to review and perhaps, add more suggestions which would be a good thing for you.

@algomaster99 
The PR title was suggested so I didn't change it.
As of branch, I was working on `algo-ooxml` itself and not custom branch. 
Last but not least, the issue is not closed completely by this PR. So I didn't mention it.

I will close this PR and create a new one from a newly created branch to algo-ooxml of upstream. 
I apologize for the inconvenience caused.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
See comments for detailed review. Make a habit of running `npx eslint . --fix` in `markdown-docx` before committing. We could configure hooks for it but let it be due to time constraints.

**Comment by algomaster99** [comment]
I just noticed you closed your PR. Anyway, do look at the reviews before submitting a new one.
